esmvalcore/experimental/config/_config_object.py

```diff
@@ -0,0 +1,191 @@
+    ----------
+    config : dict
+        Dictionary with configuration settings.
+    name : str
+        Name of the session to initialize, for example, the name of the
+        recipe (default='session').
+    """
+
+    validate = _validators
+
+    def __init__(self, config: dict, name: str = 'session'):
+        super().__init__(config)
+        self.init_session_dir(name)
+
+    def init_session_dir(self, name: str = 'session'):
+        """Initialize session.
+
+        The `name` is used to name the working directory, e.g.
+        `recipe_example_20200916/`. If no name is given, such as in an
+        interactive session, defaults to `session`.
+        """
+        now = datetime.utcnow().strftime("%Y%m%d_%H%M%S")
+        session_name = f"{name}_{now}"
+        self._session_dir = self['output_dir'] / session_name
+
+    @property
+    def session_dir(self):
+        """Return session directory."""
+        return self._session_dir
+
+    @property
+    def preproc_dir(self):
+        """Return preproc directory."""
+        return self.session_dir / 'preproc'
+
+    @property
+    def work_dir(self):
+        """Return work directory."""
+        return self.session_dir / 'work'
+
+    @property
+    def plot_dir(self):
+        """Return plot directory."""
+        return self.session_dir / 'plots'
+
+    @property
+    def run_dir(self):
+        """Return run directory."""
+        return self.session_dir / 'run'
+
+    @property
+    def config_dir(self):
+        """Return user config directory."""
+        return USER_CONFIG_DIR
+
+
+def read_config_file(config_file):
+    """Read config user file and store settings in a dictionary."""
+    config_file = Path(config_file)
+    if not config_file.exists():
+        raise IOError(f'Config file `{config_file}` does not exist.')
+
+    with open(config_file, 'r') as file:
+        cfg = yaml.safe_load(file)
+
+    return cfg
+
+
+def _load_default_config(filename: str):
+    """Load the default configuration."""
+    mapping = read_config_file(filename)
+
+    global CFG_DEFAULT
+
+    CFG_DEFAULT.update(mapping)
+
+
+def _load_user_config(filename: str, raise_exception: bool = True):
+    """Load user configuration from the given file.
+
+    The config is cleared and updated in-place.
+
+    Parameters
+    ----------
+    filename: pathlike
+        Name of the config file, must be yaml format
+    raise_exception : bool
+        Raise an exception if `filename` can not be found (default).
+        Otherwise, silently pass and use the default configuration. This
+        setting is necessary for the case where `.esmvaltool/config-user.yml`
+        has not been defined (i.e. first start).
+    """
+    try:
+        mapping = read_config_file(filename)
+        mapping['config_file'] = filename
+    except IOError:
+        if raise_exception:
+            raise
+        mapping = {}
+
+    global CFG
+
+    CFG.clear()
+    CFG.update(CFG_DEFAULT)
+    CFG.update(mapping)
```

Comment:
These functions could be methods on the `Config` class, that would be cleaner and also avoid the need for the `global` use

Reply:
As static methods you mean? I am not sure, as they are really two different things. The globals are necessary to make the configuration importable.

Reply:
> As static methods you mean?

Yes, static if needed. It would provide a better grouping of functionality

> The globals are necessary to make the configuration importable.

I meant the use of `global` statement, it's [discouraged by pylint](https://stackoverflow.com/questions/14503973/python-global-keyword-vs-pylint-w0603)

Reply:
I think I can actually get rid of the globals at this point.